monitor/utils/web3Cache.js

```diff
@@ -52,6 +58,36 @@ async function isForeignContract(address) {
   return cachedForeignIsContract[address]
 }
 
+function getPastEventsWithAPIFallback(contract, options) {
+  return commonGetPastEvents(contract, options).catch(async e => {
+    const [api, web3] =
+      options.chain === 'home' ? [MONITOR_HOME_EXPLORER_API, web3Home] : [MONITOR_FOREIGN_EXPLORER_API, web3Foreign]
+    if (api && e.message.includes('exceed maximum block range')) {
+      logger.debug('BLOCK RANGE EXCEED, using fallback to the explorer API')
+
+      const abi = contract.options.jsonInterface.find(abi => abi.type === 'event' && abi.name === options.event)
+
+      const url = new URL(api)
+      url.searchParams.append('module', 'logs')
+      url.searchParams.append('action', 'getLogs')
+      url.searchParams.append('address', contract.options.address)
+      url.searchParams.append('fromBlock', options.fromBlock)
+      url.searchParams.append('toBlock', options.toBlock || 'latest')
+      url.searchParams.append('topic0', web3.eth.abi.encodeEventSignature(abi))
+
+      const logs = await fetch(url).then(res => res.json())
```

Comment:
From the Etherscan API docs: `For performance & security considerations, only the first 1000 results are return. So please narrow down the filter parameters`.
Seems that we need to request events in batches by using some sort of pagination logic.